task/aontimer/src/main.rs

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+use aontimer as aon;
+use drv_riscv_plic_api::*;
+use userlib::*;
+
+task_slot!(INT_CONTROLLER, ext_int_ctrl);
+
+const AONTIMER_BARK: u32 = 0x1;
+const AONTIMER_TIME: u32 = 0x2;
```

Comment:
Are these notification bits posted by the plic driver? If so, then these should be suffixed with _NOTIFICATION to make it clear. Also, how does the plic know that it has to set bit 1 i.e. value 0x2? I don't see that being added to app.toml

BTW, looking at the way this gets used, it seems very brittle and redundant. app.toml needs to define the exact same notification bits which are defined here so that the task gets notified of the interrupt correctly. I think there should be just one place where this information is defined.

Reply:
Good idea with the `_NOTIFICATION` suffix.  

Within `app/demo-opentitan/app.toml`: 
```
[tasks.ext_int_ctrl.config]
ints = [158]
tasks = ['"aontimer"']
notification = [1]
base = 0x4800_0000
pbits = 3
```
Each of `ints`, `tasks`, `notification`, acts as a list where all of the elements of the same index are associated.  So this block defines that the `aontimer` task has access to interrupt `158`, using notification bit `1`.

Then in `task/aontimer/src/main.rs`, the call to `sys_set_timer` defines the notification bit to use:
```
sys_set_timer(Some(alarm + 1000), AONTIMER_TIME_NOTIFICATION);
```

The ability to extract the notification bit for the app is being tracked by this jira [ticket](https://rivosinc.atlassian.net/browse/SW-530).  Also see this [comment](https://github.com/rivosinc/hubris/pull/7#discussion_r972465239) on the initial plic PR.  I think this feature was held out for now because of some extra complications that would be needed in the build system, but @FawazTirmizi would have more context.

Reply:
After talking to @FawazTirmizi, it seems this may have exposed some bugs with the PLIC, but still have to look into it.

Reply:
Bug fixed. https://github.com/rivosinc/hubris/pull/27